brainio_base/assemblies.py

```diff
@@ -20,11 +20,17 @@ class DataAssembly(DataArray):
     """A DataAssembly represents a set of data a researcher wishes to work with for
     an analysis or benchmarking task.  """
 
+    __slots__ = ()
+
     def __init__(self, *args, **kwargs):
-        super(DataAssembly, self).__init__(*args, **kwargs)
-        gather_indexes(self)
+        temp = DataArray.__new__(DataArray, *args, **kwargs)
+        DataArray.__init__(temp, *args, **kwargs)
+        temp = gather_indexes(temp)
```

Comment:
a comment to explain what's going on / why it's necessary would be useful for future maintenance